cachito/web/models.py

```diff
@@ -590,10 +581,10 @@ def from_json(cls, kwargs):
                     # Send the changes queued up in SQLAlchemy to the database's transaction buffer.
                     # This will generate an ID that can be used below.
                     db.session.flush()
```

Comment:
I think this `db.session.flush()` is no longer needed since the code is now using the relationship. Could you please verify this?

Reply:
@tkdchen were you able to verify if this is still needed?

Reply:
@mprahl This is removed in bef4f16.

Reply:
@tkdchen This is a different instance of `db.session.flush`. Look at the diff directly and see where the comment is and it'll make more sense.

Reply:
@mprahl I removed these piece of code. What I see is SQLAlchemy flushes the changes into database automatically when using relationships.

Reply:
@tkdchen so can you remove the `db.session.flush` after all? It's still in the diff.

Same for the one below.

Reply:
@mprahl Oh, I think I misunderstood. My comment is for the flush call in `Request.add_dependency`.

Reply:
The flush call in `add_state` is still needed. Removing it will cause a CircularDependencyError. The other one is removed.